pystac_client/client.py

```diff
@@ -48,7 +49,7 @@ def open(
             catalog : A :class:`Client` instance for this Catalog/API
         """
         cat = cls.from_file(url, headers=headers, parameters=parameters)
-        search_link = cat.get_links('search')
+        search_link = cat.get_links('search', media_type=media_type)
```

Comment:
```suggestion
        search_link = cat.get_links('search', media_type=pystac.MediaType.GEOJSON)
```

Since the STAC API Spec [requires the `type` of the search link to be `"application/geo+json"`](https://github.com/radiantearth/stac-api-spec/tree/master/item-search#link-relations) it might be more appropriate to hard-code this using the [PySTAC `MediaType` enum](https://github.com/stac-utils/pystac/blob/305ed6955835183ee8a4307ae0ea9b0ef8f05fa9/pystac/media_type.py#L8).